### PR TITLE
Create Tests for Generate Report

### PR DESCRIPTION
**Pull request by nathanafsar-sigma**
This pull request adds a comprehensive test suite for the `report_generate.py` module, covering all major functions with various test cases using pytest. The tests ensure the correctness of data formatting, profit/loss calculations, user market data extraction, price chart generation, and HTML report creation.

**New test coverage for `report_generate.py`:**

* **Unit tests for core functions:** Added tests for `format_name`, `calculate_profit_loss`, `get_user_market_data`, `generate_price_chart`, and `generate_user_html_report`, verifying their behavior with multiple input scenarios.

* **Data handling and edge cases:** Tests include parameterized cases for name formatting, profit/loss calculation with valid and invalid data, and checks for handling users with no commodities.

* **Chart generation validation:** Ensures that price charts are generated only with sufficient data and handles symbols with special characters (slashes).

* **HTML report correctness:** Verifies that the generated HTML report includes expected user and commodity information, profit/loss values, and commodity counts.

### CODE REVIEW

File: daily_report/test_report_generate.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+    @pytest.mark.parametrize("input_name,expected", [
+        ("alice_jones", "Alice Jones"),
+        ("alice", "Alice"),
+        ("john_paul_jones", "John Paul Jones"),
+        ("Alice", "Alice"),
+        ("", ""),
+    ])
+    def test_format_name(self, input_name, expected):
+        assert format_name(input_name) == expected
```
[the remark this answers is not in the document]
Comment (gdownersigma) [reply daily_report/test_report_generate.py]: Copilot is right here, could we have some docstrings for each test to explain what it's testing.

**Review by gdownersigma — APPROVED** [review]

File: daily_report/test_report_generate.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nathanafsar-sigma) [reply daily_report/test_report_generate.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/gdownersigma/c21-commodities/pull/84#discussion_r2769923556)
